src/pseudopeople/noise_functions.py

```diff
@@ -149,13 +153,144 @@ def duplicate_with_guardian(
     randomness_stream: RandomnessStream,
 ) -> pd.DataFrame:
     """
-
-    :param dataset_name:
-    :param dataset_data:
-    :param configuration:
-    :param randomness_stream:
+    Function that duplicates rows of a dataset. Rows that are duplicated fall into one of three groups of
+    dependents that are typically duplicated in administrative data where a dependent lives in a different
+    location then their guardian and they show up in the data at both locations. For simplicity, rows will
+    be only duplicated once for a maximum of two rows per dependent. When a row is duplicated, one row will
+    have the depdent's correct address and the other will have the guardian's address.
+    :param dataset_name: Name of the dataset being noised
+    :param dataset_data: pd.DataFrame that will be noised
+    :param configuration: ConfigTree object containing noise level values. Dict with 3 key groups for duplication
+    :param randomness_stream: RandomnessStream instance to make random selection for noise
     :return:
     """
+
+    # dataset_data = dataset_data.copy()
+    # Get dict of group type and formatted dataframe for that group that should be noised
+    formatted_group_data = {}
+    # Get dataframe for each dependent group to merge with guardians
+    in_households_under_18 = dataset_data.loc[
+        (dataset_data["age"] < 18)
+        & (dataset_data["housing_type"] == "Household")
+        & (dataset_data["guardian_1"].notna())
+    ]
+    in_households_18_to_23 = dataset_data.loc[
+        (dataset_data["age"] >= 18)
+        & (dataset_data["age"] < 24)
+        & (dataset_data["housing_type"] == "Household")
+        & (dataset_data["guardian_1"].notna())
+    ]
+    in_group_quarters_under_24 = dataset_data.loc[
+        (dataset_data["age"] < 24)
+        & (dataset_data["housing_type"] != "Household")
+        & (dataset_data["guardian_1"].notna())
+    ]
+
+    # Helper function to format group dataframe and merging with their dependentss
+    def _merge_depdents_and_guardians(
```

Comment:
nit: move this to the top. This might just be my personal preference as opposed to any sort of "rule" though so nbd either way